libr/anal/anal.c

```diff
@@ -213,12 +213,27 @@ R_API bool r_anal_use(RAnal *anal, const char *name) {
 }
 
 R_API char *r_anal_get_reg_profile(RAnal *anal) {
-	return (anal && anal->cur && anal->cur->get_reg_profile)
+	r_return_val_if_fail (anal, NULL);
+	if (anal->arch && anal->arch->cur && anal->arch->cur->registers) {
```

Comment:
This part might become a macro too, since will be the same across many r2 plugins